yesod-auth-oauth2.cabal

```diff
@@ -36,6 +36,7 @@ library
                    , aeson                   >= 0.6       && < 0.9
                    , yesod-core              >= 1.2       && < 1.5
                    , authenticate            >= 1.3.2.7   && < 1.4
+                   , random
```

Comment:
Is there a reason to add this and do what you're doing vs just moving what the GitHub plugin was doing with `UUID.nextRandom` into the base plugin? If so, can we remove a `uuid` dependency here too?